developer-hub/src/reference_docs/glossary.rs

```diff
@@ -36,41 +36,71 @@
 //!
 //! #### Offchain
 //!
+//! Refers to operations conducted outside the blockchain's consensus mechanism. They are essential
+//! for enhancing scalability and efficiency, enabling activities like data fetching and computation
+//! without bloating the blockchain.
+//!
 //! #### Host Function:
 //!
+//! Host functions are functions provided by the runtime environment (the [host](#host)) to the Wasm
+//! runtime. These functions allow the Wasm code to interact with and perform operations on the
+//! [node](#node), like accessing the blockchain state.
+//!
 //! #### Runtime API:
 //!
+//! The runtime API acts as a communication bridge between the runtime and the node, serving as the
+//! exposed interface that facilitates their interactions.
+//!
 //! #### Dispatchable:
 //!
-//! Callable
+//! Dispatchables are functions that can be called by external entities, such as users or external
+//! systems, to interact with the blockchain's state. They are a core aspect of the runtime logic,
+//! handling transactions and other state-changing operations.
+//!
+//! **Synonyms**: Callable
 //!
 //! #### Extrinsic
 //!
+//! An extrinsic is a general term for a piece of data that is originated outside of the runtime and
+//! fed into the it as a part of the block-body. This includes user-initiated transactions as well
+//! as inherents which are placed into the block by the block-builder.
 //!
 //! #### Pallet
 //!
-//! #### Full Node
-//!
-//! #### Archive Node
-//!
-//! #### Validator
-//!
-//! #### Collator
-//!
-//! #### Parachain
-//!
-//! aka. AppChain.
-//!
-//!
-//!
+//! FRAME pallets are modular components that encapsulate specific functionalities or
+//! business logic of a blockchain. They are the building blocks used to construct a blockchain's
+//! runtime, allowing for customizable and upgradeable networks. Pallets can be used to extend the
+//! capabilities of a Substrate-based blockchain in a composable way.
 //!
+//! #### Full Node
 //!
+//! It is a node that prunes historical states, keeping only recent finalized block states to reduce
```

Comment:
Not sure if this is always the case. A full node that prunes is normally called a pruned node. A full node can also not prune, but discard the preimage data (like the user transactions) and still have all block data.  
But maybe im wrong, cc @paritytech/sdk-node

Reply:
I got it from here https://wiki.polkadot.network/docs/maintain-sync#:~:text=A%20full%20node%20prunes%20historical,space%20than%20an%20archive%20node.

Reply:
We are calling them full nodes and archive nodes, where archive nodes are full nodes that have pruning disabled and store historical data.